### ISSUE TITLE
Bug accounting for modes of RFFT arrays

### ISSUE THREAD

**Issue by michael-0brien** [mjo22/cryojax#563]
There are a few places in the code with a bug in how we account for RFFT modes. Let me give an example:

Consider the task of computing the L2 norm from an RFFT array `f`, which is shape `N//2+1` and shape `N` in real space. `f[0]` represents one mode for the full size `N` array, while `f[1:]` represents 2 modes each. When computing the L2 norm, the correct form is `l2 = jnp.abs(f[0])**2 + 2 * jnp.sum(jnp.abs(f[1:])**2)` (in practice, we should implement as `l2 = jnp.sum(mask * f * jnp.conj(f))`). If `N` is even, `f[-1]` also has multiplicity of 1 due to periodicity. I won’t write the generalization to high dimensions here but basically the same holds but just on the last dimension of the array.

We definitely have a few bugs in the code in this regard, but thankfully it’s small and doesn’t impact most use cases. Here are the places I can think of:

- Power spectra + FSC/FRCs: https://github.com/michael-0brien/cryojax/blob/dev/cryojax/ndimage/_fourier_statistics.py#L13
- Normalizing images given in Fourier domain: https://github.com/michael-0brien/cryojax/blob/dev/cryojax/ndimage/_normalize.py#L79
- Likelihood in Fourier domain: https://github.com/michael-0brien/cryojax/blob/dev/cryojax/simulator/_noise_model/gaussian_noise_model.py#L347

**Comment by michael-0brien** [comment]
Will be closed in #579
